### PR TITLE
Add an Id class to make sessions easier to deal with.

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/id.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import assert from '../platform/assert-web.js';
+
+ export default class Id {
```
Comment (dstockwell) [inline runtime/id.js]: indentation is weird

File: runtime/id.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    return id;
+  }
+
+  toString() {
+    return `!${this.session}:${this.components.join(':')}`;
+  }
+
+  // Only use this for testing!
```
Comment (dstockwell) [inline runtime/id.js]: add `ForTesting` too the name, `toStringForTesting`?

File: runtime/id.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import assert from '../platform/assert-web.js';
+
+ export default class Id {
+  constructor(currentSession) {
+    this.session = currentSession;
```
Comment (dstockwell) [inline runtime/id.js]: why not _session etc?

File: runtime/id.js (hunk 1 above)
Comment (shans) [reply runtime/id.js]: Done

File: runtime/id.js (hunk 2 above)
Comment (shans) [reply runtime/id.js]: Done

File: runtime/id.js (hunk 3 above)
Comment (shans) [reply runtime/id.js]: Done

**Pull request merged**